lib/rules/no-else-return.js

```diff
@@ -33,7 +35,45 @@ module.exports = {
          * @returns {void}
          */
         function displayReport(node) {
-            context.report({ node, message: "Unnecessary 'else' after 'return'." });
+            context.report({
+                node,
+                message: "Unnecessary 'else' after 'return'.",
+                fix: fixer => {
+                    const sourceCode = context.getSourceCode();
+                    const startToken = sourceCode.getTokenByRangeStart(node.start);
+                    const elseToken = sourceCode.getTokenBefore(startToken);
+                    const source = sourceCode.getText(node);
+                    const lastIfToken = sourceCode.getTokenBefore(elseToken);
+                    let fixedSource;
+
+                    if (startToken.type === "Punctuator" && startToken.value === "{") {
+                        const firstTokenOfElseBlock = sourceCode.getTokenAfter(startToken);
+
+                        if (lastIfToken.value !== "}" &&
```

Comment:
Good point -- two potential workarounds I can think of:

* In [this](https://github.com/xdumaine/eslint/blob/f4f8ac4894b2fd433bfb558714ce2a36d8b0da81/lib/rules/no-else-return.js#L190) call to `displayReport`, pass `node` instead of `alternate` (and then access the property accordingly within the `displayReport` function to make sure everything else still works). That way, `node` will refer to the `IfStatement`, so you could check `node.consequent.type`.
* Change [this listener](https://github.com/xdumaine/eslint/blob/f4f8ac4894b2fd433bfb558714ce2a36d8b0da81/lib/rules/no-else-return.js#L170) to `"IfStatement:exit"` instead of `IfStatement`

Explanation: The `parent` property is added to each node as ESLint enters it while traversing. Since this listener is fired when the `IfStatement` node is entered, the `parent` property hasn't been added to the `else` node yet. If the listener is fired when the `IfStatement` node is exited instead, the `else` node will have its parent property.